src/asciidocParser.ts

```diff
@@ -5,24 +5,21 @@ import { isNullOrUndefined } from 'util';
 
 const fileUrl = require('file-url');
 
-const asciidoctor = require('@asciidoctor/core')()
+const asciidoctor = require('@asciidoctor/core')
 const docbook = require('@asciidoctor/docbook-converter')
 const kroki = require('asciidoctor-kroki')
 const highlightjsAdapter = require('./highlightjs-adapter')
-highlightjsAdapter.register(asciidoctor)
-
-const use_kroki = vscode.workspace.getConfiguration('asciidoc', null).get('use_kroki');
-if (use_kroki) {
-  kroki.register(asciidoctor.Extensions);
-}
 
 export class AsciidocParser {
     public html: string = '';
     public document = null;
+    public processor = null
+    public registry = null
```

Comment:
I don't think we need to declare the `registry` as a class attribute.
It should be enough to declare a local variable in the `convert_using_javascript` function no?